plugins/woocommerce/templates/emails/customer-completed-order.php

```diff
@@ -29,8 +29,16 @@
 do_action( 'woocommerce_email_header', $email_heading, $email ); ?>
 
 <?php echo $email_improvements_enabled ? '<div class="email-introduction">' : ''; ?>
-<?php /* translators: %s: Customer first name */ ?>
-<p><?php printf( esc_html__( 'Hi %s,', 'woocommerce' ), esc_html( $order->get_billing_first_name() ) ); ?></p>
+<p>
+<?php
+if ( ! empty( $order->get_billing_first_name() ) ) {
+	/* translators: %s: Customer first name */
+	printf( esc_html__( 'Hi %s,', 'woocommerce' ), esc_html( $order->get_billing_first_name() ) );
+} else {
+	printf( esc_html__( 'Hi,' ) );
```

Comment:
@SpaceSurfer1 one last thing, the CI lint action is failing (see [here](https://github.com/woocommerce/woocommerce/actions/runs/13409024200/job/37456865196#step:4:27)) because the `esc_html__` function needs a domain parameter as a second parameter, it should be:
```php
printf( esc_html__( 'Hi,', 'woocommerce' ) );
```
Thanks!